marlowe-runtime/web-server/Language/Marlowe/Runtime/Web/Server/Flags.hs

```diff
@@ -0,0 +1,13 @@
+{-# LANGUAGE GADTs #-}
+
+module Language.Marlowe.Runtime.Web.Server.Flags
+  where
+
+data Enabled
+data Disabled
+
+data Flag a where
+  Enabled :: Flag Enabled
+  Disabled :: Flag Disabled
+
+data SomeFlag = forall a. SomeFlag (Flag a)
```

Comment:
What is the benefit of this approach over just doing
```haskell
data FlagEnabled = Enabled
data FlagDisabled = Disabled
```
?
Also it seems like they're not used anywhere; remove?

Reply:
Yes, was going to remove it, thanks! The idea was to have a kind-level representation and use a type family to make the API type depend on the flag value, e.g.

```hs
type family API openApiEnabled where
  API Disabled = Web.API
  API Enabled = OpenAPI.Api :<|> Web.API

server :: Flag openAPIEnabled -> Server (Api openAPIEnabled)
server openAPIFlag = case openAPIFlag of
  Enabled -> OpenAPI.server :<|> Web.API
  Disabled -> Web.API
```

But I chose a different, simpler approach in the end.